server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -712,19 +712,17 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
 
     }
 
-    public SourceLoader.SyntheticFieldLoader syntheticFieldLoader(Stream<Mapper> extra) {
-        return new SyntheticSourceFieldLoader(
-            Stream.concat(extra, mappers.values().stream())
-                .sorted(Comparator.comparing(Mapper::name))
-                .map(Mapper::syntheticFieldLoader)
-                .filter(l -> l != null)
-                .toList()
-        );
+    public SourceLoader.SyntheticFieldLoader syntheticFieldLoader(Stream<Mapper> mappers) {
+        var fields = mappers.sorted(Comparator.comparing(Mapper::name))
+            .map(Mapper::syntheticFieldLoader)
+            .filter(l -> l != SourceLoader.SyntheticFieldLoader.NOTHING)
```

Comment:
No `SyntheticFieldLoader` returns `null`, so instead we should exclude the ones that return `NOTHING`. Most meta field mappers return `NOTHING`, except `DocCountFieldMapper`.